packages/nimble-components/src/rich-text-editor/specs/README.md

```diff
@@ -128,7 +131,22 @@ application's performance is enhanced as the operation is performed only once, t
 
 _Methods_
 
--   none
+-   `hideFooter()` - hides the footer section that contains all formatting options and the `footer-actions` slot.
+-   `clearContent()` - clears the content in the editor through [clearContent()](https://tiptap.dev/api/commands/clear-content) command in Tiptap.
```

Comment:
How is this different than assigning `el.markdown = ''` or `el.markdown = undefined;`?

Reply:
Initially, our plan was to clear the content without modifying the markdown value in the component. We considered this for scenarios like clicking a cancel button while adding a comment, where we preferred not to change the markdown value since it had not been parsed yet. Additionally, the reason for updating the markdown value was not entirely evident, and considering that Tiptap offers a method for clearing content, we decided to expose this API.

However, if we set the markdown value to an empty string, we can clear the content from the client. As this functionality can be achieved directly using the markdown and to avoid duplicating the same functionality in two different places, I have currently removed this method.

Reply:
@rajsite To reconfirm that this approach works in all cases, we did a second pass check of the actual use case of the `clearContent()` method API and found that it might be required in a few instances. 

For example, in comments, if the user adds a new comment and clicks the `Cancel` button, the editor will not be cleared since Angular's change detection is not triggered. This happens because the editor's `el.markdown` value remains unchanged (i.e., the markdown value does not update when the user doesn't click the `Ok` button. Only upon clicking `Ok`, the markdown value changes, not with every input). 

With the `clearContent()` method exposed, on click of the `Cancel` button, we can clear the content from the editor without changing the markdown value. So we would like to bring back this method as it gives a valid use case that may not be achieved by setting the `el.markdown = ''`. Please let us know your thoughts.

Reply:
We ran into the same behavior with the nimble-table and the data property. Angular's change detection would say the value was not updated if we re-assigned the same data reference even if the data was changed.

I'm betting if in Angular you did not rely on data binding but instead used ViewChild and got the component reference and assigned directly to the directive it would work. But I can see how relying on data binding would cause issues with change detection.

I think this is a good indication that for the rich text editor we should use explicit getMarkdown and setMarkdown methods instead of relying on accessors and data binding.

We should not change the rich text viewer API, it makes sense as a property with the convenience of data binding. But it seems like for reasons similar to the table we should use methods for the rich text editor instead. I'd suuggest we change to getMarkdown and setMarkdown instead of adding a clearContent method.